src/number_theory/legendre_symbol/quadratic_reciprocity.lean

```diff
@@ -62,43 +74,27 @@ begin
     refine ⟨units.mk0 y hy, _⟩, simp, }
 end
 
-lemma exists_sq_eq_neg_one_iff : is_square (-1 : zmod p) ↔ p % 4 ≠ 3 :=
-begin
-  have h := @finite_field.is_square_neg_one_iff (zmod p) _ _,
-  rw card p at h,
-  exact h,
-end
-
-lemma mod_four_ne_three_of_sq_eq_neg_one {y : zmod p} (hy : y ^ 2 = -1) : p % 4 ≠ 3 :=
-begin
-  rw pow_two at hy,
-  exact (exists_sq_eq_neg_one_iff p).1 ⟨y, hy.symm⟩
-end
-
-lemma mod_four_ne_three_of_sq_eq_neg_sq' {x y : zmod p} (hy : y ≠ 0) (hxy : x ^ 2 = - y ^ 2) :
-  p % 4 ≠ 3 :=
-@mod_four_ne_three_of_sq_eq_neg_one p _ (x / y) begin
-  apply_fun (λ z, z / y ^ 2) at hxy,
-  rwa [neg_div, ←div_pow, ←div_pow, div_self hy, one_pow] at hxy
-end
-
-lemma mod_four_ne_three_of_sq_eq_neg_sq {x y : zmod p} (hx : x ≠ 0) (hxy : x ^ 2 = - y ^ 2) :
-  p % 4 ≠ 3 :=
-begin
-  apply_fun (λ x, -x) at hxy,
-  rw neg_neg at hxy,
-  exact mod_four_ne_three_of_sq_eq_neg_sq' p hx hxy.symm
-end
-
+/-- If `a : zmod p` is nonzero, then `a^(p/2)` is either `1` or `-1`. -/
 lemma pow_div_two_eq_neg_one_or_one {a : zmod p} (ha : a ≠ 0) :
   a ^ (p / 2) = 1 ∨ a ^ (p / 2) = -1 :=
 begin
-  cases nat.prime.eq_two_or_odd (fact.out p.prime) with hp2 hp_odd,
+  cases prime.eq_two_or_odd (fact.out p.prime) with hp2 hp_odd,
   { substI p, revert a ha, dec_trivial },
   rw [← mul_self_eq_one_iff, ← pow_add, ← two_mul, two_mul_odd_div_two hp_odd],
   exact pow_card_sub_one_eq_one ha
 end
 
+end euler
+
+section legendre
+
+/-!
+### Definition of the Legendre symbol and basic properties
+-/
+
+variables (p : ℕ) [hpp : fact p.prime]
+include hpp
```

Comment:
Same here; should not be necessary to name `hpp`.

Reply:
This instance is used explicitly in the proof of `legendre_sym_eq_pow`.
Of course, one could repeat and name the instances in the statements that use them explicitly and have them unnamed for everything else. I'm not sure what the preferred style is in this respect.

Reply:
Since you only explicitly use it once, you can replace `hpp` by the slightly longer `fact.out p.prime`.

Reply:
OK.